### PR TITLE
Store results and tags in firebase

### PR DESCRIPTION
**Pull request by ivanzvonkov**
**Changes**
- Added code to save all prediction info to firebase from inference container
- Added code to upload some percentage of the result images (with bounding boxes)
- Restructured project a bit to make building docker image more straightforward (specifically copying over street2sat_utils)
- Deleted .txt files no longer needed

**Notes**
- #15 must be merged first
- Was able to do inference on 5600 images in about ~4 mins

### CODE REVIEW

File: gcp/inference/handler.py [hunk 1]
```diff
@@ -94,63 +105,69 @@ def preprocess(
...
+            )
+            cv2.imwrite(
+                str(local_dest_path), cv2.cvtColor(labeled_img, cv2.COLOR_RGB2BGR)
+            )
+            labeled_uri = save_to_bucket(uri_as_path, local_dest_path)
+            save_to_db["labeled_img"] = labeled_uri
+
+        collection = "street2sat"
+        print(f"HANDLER: Saving to collection: {collection}, document: {name}")
+        db.collection(collection).document(pred.name).set(save_to_db)
```
Comment (ivanzvonkov) [inline gcp/inference/handler.py:163]: Saves data to firestore

File: gcp/inference/handler.py [hunk 2]
```diff
@@ -94,63 +105,69 @@ def preprocess(
...
+        uri, img, results, tags = data
+        uri_as_path = Path(uri)
+        name = "-".join(uri_as_path.parts[2:-1]) + "-" + uri_as_path.stem
+        pred = Prediction.from_results_and_tags(
+            results=results, tags=tags, name=name, img_np=img
         )
+        save_to_db = pred.to_dict()
+        save_to_db["input_img"] = uri
+
+        labeled_uri = None
```
Comment (ivanzvonkov) [inline gcp/inference/handler.py:149]: If `LABEL_IMG_PERCENT=0.01` then 1% of all images will have their labeled image (with bounding boxes) uploaded to a bucket. 

If we upload a labeled image for every single input image that would be a big waste of time and space on the cloud.

File: street2sat_utils/client.py [hunk 3]
```diff
@@ -1,32 +1,49 @@
...
+    return helper
 
-model: Model = hubconf.custom(str(MODEL_PATH))
-model.eval()
+
+@memoize
+def load_model(model_path: str):
+    model = hubconf.custom(model_path)
+    model.eval()
+    return model
```
Comment (ivanzvonkov) [inline street2sat_utils/client.py:40]: Added a memoized function for loading the model so it's only loaded when needed and only loaded once.

File: street2sat_utils/client.py [hunk 4]
```diff
@@ -1,32 +1,49 @@
-from __future__ import annotations
```
Comment (ivanzvonkov) [inline street2sat_utils/client.py:1]: Due to the torchserve Docker container using python3.6 this cannot be used

File: README.md [hunk 5]
```diff
@@ -47,6 +46,30 @@ gsutil mb gs://street2sat-model-predictions
...
+{'time': DatetimeWithNanoseconds(2021, 6, 23, 13, 53, 42, tzinfo=<UTC>),
+ 'distances': {'maize': 16.36581243763188},
+ 'focal_length': 3,
+ ...
+
+>> query = coll.where("name", ">=", "Uganda-2021-06-23_lutoto_left2-100GOPRO")
+            .where("name", "<", "Uganda-2021-06-23_lutoto_left2-101GOPRO").get()
+>> len(q)
+999
+```
```
Comment (ivanzvonkov) [inline README.md:71]: The above is a demo of what will be available after this is merged

**Review by mangomadhava — COMMENTED** [review]
nice

**Review by mangomadhava — APPROVED** [review]

**Pull request merged**